spark-submission-worker/src/main/java/org/apache/spark/k8s/operator/SparkClusterResourceSpec.java

```diff
@@ -271,4 +277,55 @@ private static StatefulSet buildWorkerStatefulSet(
         .endSpec()
         .build();
   }
+
+  private static Optional<HorizontalPodAutoscaler> buildHorizontalPodAutoscaler(
+      String clusterName, String namespace, ClusterSpec spec) {
+    var instanceConfig = spec.getClusterTolerations().getInstanceConfig();
+    if (instanceConfig.getMinWorkers() >= instanceConfig.getMaxWorkers()) {
+      return Optional.empty();
+    }
+    return Optional.of(
+        new HorizontalPodAutoscalerBuilder()
+            .withNewMetadata()
+            .withNamespace(namespace)
+            .withName(clusterName + "-worker-hpa")
+            .endMetadata()
+            .withNewSpec()
+            .withNewScaleTargetRef()
+            .withApiVersion("apps/v1")
+            .withKind("StatefulSet")
+            .withName(clusterName + "-worker")
+            .endScaleTargetRef()
+            .withMinReplicas(instanceConfig.getMinWorkers())
+            .withMaxReplicas(instanceConfig.getMaxWorkers())
+            .addToMetrics(
+                new MetricSpecBuilder()
+                    .withType("Resource")
+                    .withNewResource()
+                    .withName("cpu")
+                    .withNewTarget()
+                    .withType("Utilization")
+                    .withAverageUtilization(30)
+                    .endTarget()
+                    .endResource()
+                    .build())
```

Comment:
Can these configs like resource and average utilization to be specify in cluster yaml file instead hard-coded?

Reply:
Ya, template support is the next step, @viirya .

Reply:
Okay. Looking forward to the next steps. Thanks.